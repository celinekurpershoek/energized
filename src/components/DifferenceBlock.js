import React from "react";
export function DifferenceBlock(props) {
  console.log(props);
  if (!props.percentage) {
    return null;
  }
  return (
    <span
      title={`${props.difference}KWH (+${props.percentage}%)`}
      className="graph__pipe graph__pipe--vertical graph__pipe--difference"
      style={{
        height: `${props.percentage}%`
      }}
      hidden={props.percentage < 1}
    />
  );
}
