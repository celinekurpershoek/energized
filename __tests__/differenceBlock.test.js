import React from "react";
import renderer from "react-test-renderer";
import DifferenceBlock from "../src/components/differenceBlock";

it("Correctly render header component", () => {
  const component = renderer.create(
    <DifferenceBlock difference="10" percentage="50" />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
