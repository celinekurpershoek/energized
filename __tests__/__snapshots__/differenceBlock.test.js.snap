// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly render differenceBlock component 1`] = `
<span
  className="graph__pipe graph__pipe--vertical graph__pipe--difference"
  style={
    Object {
      "height": "50%",
    }
  }
  title="10KWH (+50%)"
/>
`;
